misc/python/materialize/scalability/endpoint.py

```diff
@@ -46,13 +46,21 @@ def sql(self, sql: str) -> None:
         cursor = conn.cursor()
         cursor.execute(sql.encode("utf8"))
 
-    def name(self) -> str:
-        if self._name is not None:
-            return self._name
+    def try_load_version(self) -> str:
+        """
+        Tries to load the version from the database or returns 'unknown' otherwise.
```

Comment:
Under what circumstances would `unknown` be returned?

Reply:
The problem is that you can't invoke this method at any time. That's also what caused the error. The endpoint needs to be up for this to work. That is, if you invoke this method too early (before having executed `endpoint.up()`), the query will fail.

Reply:
I see.

Reply:
The same happened if you invoked it too late on an earlier version when the endpoint was already down. I had fixed this by caching the information once I have it.